go/service/user.go

```diff
@@ -344,6 +345,248 @@ func (h *UserHandler) UploadUserAvatar(ctx context.Context, arg keybase1.UploadU
+	PickerText    string                // "Twitter", "Your own website", "octodon.xyz"
+	PickerSubtext string                // "twitter.com", "Mastodon instance"
+	PickerIcon    []keybase1.SizedImage // xxx fill these in
+	Priority      int
+}
+
+func (h *UserHandler) profileProofSuggestionsHelper(mctx libkb.MetaContext) (ret []ProofSuggestion, err error) {
+	user, err := libkb.LoadMe(libkb.NewLoadUserArgWithMetaContext(mctx).WithPublicKeyOptional())
+	if err != nil {
+		return ret, err
+	}
+	if user == nil {
+		return ret, fmt.Errorf("could not load logged-in user")
+	}
+
+	var dummyIcon []keybase1.SizedImage // TODO CORE-9882: Get some icons
+	var suggestions []ProofSuggestion
+	ossifiedSocial := map[string]ProofSuggestion{
+		"github": {
+			Key:           "github",
+			ProfileText:   "Prove your GitHub",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "GitHub",
+			PickerSubtext: "github.com",
+			PickerIcon:    dummyIcon,
+		},
+		"hackernews": {
+			Key:           "hackernews",
+			ProfileText:   "Prove your Hacker News",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Hacker News",
+			PickerSubtext: "news.ycombinator.com",
+			PickerIcon:    dummyIcon,
+		},
+		"reddit": {
+			Key:           "reddit",
+			ProfileText:   "Prove your Reddit",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Reddit",
+			PickerSubtext: "reddit.com",
+			PickerIcon:    dummyIcon,
+		},
+		"twitter": {
+			Key:           "twitter",
+			ProfileText:   "Prove your Twitter",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Twitter",
+			PickerSubtext: "twitter.com",
+			PickerIcon:    dummyIcon,
+		},
+		"rooter": {
+			Key:           "rooter",
+			ProfileText:   "Prove your Rooter",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Rooter",
+			PickerSubtext: "",
+			PickerIcon:    dummyIcon,
+		}}
+	serviceKeys := mctx.G().GetProofServices().ListServicesThatAcceptNewProofs()
+	for _, service := range serviceKeys {
+		switch service {
+		case "web", "dns", "http", "https":
+			// These are under the "web" umbrella.
+			// "web" is added below.
+			continue
+		}
+		serviceType := mctx.G().GetProofServices().GetServiceType(service)
+		if serviceType == nil {
+			mctx.CDebugf("missing proof service type: %v", service)
+			continue
+		}
+		if len(user.IDTable().GetActiveProofsFor(serviceType)) > 0 {
+			mctx.CDebugf("user has an active proof: %v", serviceType.Key())
+			continue
+		}
+		if suggestion, ok := ossifiedSocial[service]; ok {
+			// Ignore the server and use hardcoded markup.
+			suggestions = append(suggestions, suggestion)
+		} else {
+			suggestions = append(suggestions, ProofSuggestion{
+				Key:           service,
+				ProfileText:   fmt.Sprintf("Prove your %v", serviceType.DisplayName()),
+				ProfileIcon:   dummyIcon,
+				PickerText:    serviceType.DisplayName(),
+				PickerSubtext: serviceType.GetTypeName(), // xxx url or mastodon
+				PickerIcon:    dummyIcon,
+			})
+		}
+	}
+	hasPGP := len(user.GetActivePGPKeys(true)) > 0
```

Comment:
i think these and the web/crypto ones should come from the server as well

Reply:
The user should only see a suggestion to prove pgp if they don't have a pgp key. Here that's determined from the loaded user.

Reply:
we should modify the map based on this value but i think we should drop all of the interspersed hardcoded data